### PR TITLE
chore: support PROTOBUF for PRINT TOPIC

### LINKED ISSUE #4516: Add Protobuf support to PRINT TOPIC
When implementing PROTOBUF support (#4469) I originally had added support for PRINT TOPIC commands, but with the changes going on around #4507 I removed it to allow that jitter to settle down. We should add it back when that work is done.

### PR DESCRIPTION
**Pull request by agavra**
fixes #4516 

### Description 

Adds support for `PRINT TOPIC` for protobuf topics

### Testing done 

Unit testing and manual:

```
ksql> PRINT print_proto FROM BEGINNING;
Key format: UNDEFINED
Value format: PROTOBUF
rowtime: 2/19/20 9:40:02 AM PST, key: <null>, value: COL1: "a" COL2: 2 COL3: "c"
^CTopic printing ceased
```

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by hjafarpour** [comment]
```
Key format: UNDEFINED
Value format: PROTOBUF
```

@agavra usually Key and Value had the same format when using schema registry. What does `UNDEFINED` mean in this context? Doe we show `UNDEFINED` if the key format is PROTOBUF?

**Comment by agavra** [comment]
@hjafarpour  - non-KAFKA keys are not supported at all yet. It is `UNDEFINED` since all of the keys so far were `null`. If it was `long` or `int` or `string` it would've shown that value.

**Comment by hjafarpour** [comment]
Thanks @agavra , can't wait to have AVRO/JSON/PROTOBUF key support! ;)

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/RecordFormatter.java [hunk 1]
```diff
@@ -248,7 +251,7 @@ String format(final Bytes bytes) {
     ) {
       try {
         final Object result = deserializer.deserializer.deserialize(topicName, bytes.get());
-        return Optional.of(result == null ? "<null>" : result.toString());
+        return Optional.of(result == null ? "<null>" : deserializer.format(result));
```
Comment (big-andy-coates) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/RecordFormatter.java]: This call to `format` seems completely superfluous.  This method now calls `deserialize` on the deserializer and then immediately passes the result back  to `format` on the same deserializer.  Why not just have `deserialize` return the formatted result?

In the case of the new protobuf format, all we need to do is pass a factory method to the constructor of the `PROTOBUF` enum that wraps the `KafkaProtobufDeserializer` instance in one that applies the format, in a similar way to `newJsonDeserializer`.

```java
   private static Deserializer<?> newProtobufDeserializer(final SchemaRegistryClient srClient) {

    final Printer printer = TextFormat.printer();
    final KafkaProtobufDeserializer<?> inner = new KafkaProtobufDeserializer<>(srClient);
    
    return (Deserializer<Object>) (topic, data) -> {
      final Message msg = inner.deserialize(topic, data);
      if (msg == null) {
        return null;
      }
      
      return printer.shortDebugString(msg);
    };
  }

...

 enum Format {
    AVRO(0, KafkaAvroDeserializer::new),
    PROTOBUF(0, RecordFormatter::newAvroDeserializer),
    JSON(RecordFormatter::newJsonDeserializer),
 ...
 } 
```

This avoids the need to complicate the code with another function call on _all_ deserializers...

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/RecordFormatter.java [hunk 2]
```diff
@@ -257,39 +260,59 @@ String format(final Bytes bytes) {
...
     }
   }
 
   enum Format {
-    AVRO(0, KafkaAvroDeserializer::new),
+    AVRO(0, KafkaAvroDeserializer::new, Object::toString),
+    PROTOBUF(
+        0,
+        KafkaProtobufDeserializer::new,
+            o -> TextFormat.printer().shortDebugString((Message) o)),
```
Comment (big-andy-coates) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/RecordFormatter.java]: Off-topic, but just wanted to check you have it on your radar to split out a `SR_JSON` out of the `JSON` format?

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/streaming/RecordFormatterTest.java [hunk 3]
```diff
@@ -526,6 +547,95 @@ public void shouldFormatValidSessionWindowedAvro() {
...
+
+      // When:
+      deserializers.format(SERIALIZED_PROTOBUF_RECORD);
+
+      // Then:
+      assertThat(deserializers.getPossibleFormats(), not(hasItem("PROTOBUF")));
+    }
+
+    @Test
+    public void shouldNotExcludeAvroOnValidProtobuf() {
```
Comment (big-andy-coates) [inline ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/streaming/RecordFormatterTest.java]:
```suggestion
    public void shouldNotExcludeProtobufOnValidProtobuf() {
```

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/streaming/RecordFormatterTest.java [hunk 4]
```diff
@@ -526,6 +547,95 @@ public void shouldFormatValidSessionWindowedAvro() {
...
+
+      // When:
+      final String formatted = deserializers.format(SERIALIZED_PROTOBUF_RECORD);
+
+      // Then:
+      assertThat(formatted, is("str1: \"My first string\" str2: \"My second string\""));
+    }
+
+    @Test
+    public void shouldNotExcludeTimeWindowedProtobufOnValidTimeWindowedAvro() {
```
Comment (big-andy-coates) [inline ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/streaming/RecordFormatterTest.java]:
```suggestion
    public void shouldNotExcludeTimeWindowedProtobufOnValidTimeWindowedProtobuf() {
```

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/streaming/RecordFormatterTest.java [hunk 5]
```diff
@@ -526,6 +547,95 @@ public void shouldFormatValidSessionWindowedAvro() {
...
+      givenProtoSchemaRegistered();
+
+      // When:
+      final String formatted = deserializers.format(SERIALIZED_TIME_WINDOWED_PROTOBUF_RECORD);
+
+      assertThat(formatted, is("[str1: \"My first string\" str2: \"My second string\"@1234567890123/-]"));
+    }
+
+    @Test
+    public void shouldNotExcludeSessionWindowedProtobufOnValidTimeWindowedAvro() {
```
Comment (big-andy-coates) [inline ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/streaming/RecordFormatterTest.java]:
```suggestion
    public void shouldNotExcludeSessionWindowedProtobufOnValidTimeWindowedProtobuf() {
```

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/RecordFormatter.java (hunk 2 above)
Comment (big-andy-coates) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/RecordFormatter.java]: Found it: https://github.com/confluentinc/ksql/pull/4596

File: ksql-engine/src/main/java/io/confluent/ksql/engine/InsertValuesExecutor.java [hunk 6]
```diff
@@ -453,6 +456,7 @@ private static SqlType columnType(final ColumnName column, final LogicalSchema s
         }
       }
 
+      LOG.error("Could not serialize row.", e);
```
[the remark this answers is not in the document]
Comment (agavra) [reply ksql-engine/src/main/java/io/confluent/ksql/engine/InsertValuesExecutor.java]: the calling code doesn't log it - it gets returned to the user in the CLI as "could not serialize row" but it doesn't give the full trace. This has bit me several times so I'd rather keep it here as at some point we should log it (we can debate where, I don't care that much)

**Pull request merged**